src/wasm.h

```diff
@@ -1074,7 +1072,8 @@ class Host : public SpecificExpression<Expression::HostId> {
   Name nameOperand;
   ExpressionList operands;
 
-  void finalize();
+  void finalize(Type type_);  // Pass pointer size.
+  void finalize();  // FIXME: wasm32 only.
```

Comment:
I think it might be best to remove this wasm32-only variant?

Reply:
That would break binaryen.js.. I added it back in to support that. See my comment above:

Ok, changed the way Host::finalize works and is called to satisfy the tests, but it is very broken. What really needs to happen is for it to use the current pointer size, which depends on the current memory/module, which is not accessible from most of these functions calling finalize. We could stick an is64 bool in the visitors calling this function, e.g. BinaryenExpressionFinalize would need to take a module argument, changing the JS API etc.

We can do that in this PR, or something to be done in the future?